compiler/rustc_parse/src/parser/generics.rs

```diff
@@ -146,16 +147,14 @@ impl<'a> Parser<'a> {
                             (None, Vec::new())
                         };
 
-                        if this.eat(&token::Eq) {
-                            if this.check_lifetime() {
-                                let _lifetime = this.expect_lifetime();
-                                this.struct_span_err(
-                                    this.prev_token.span,
-                                    "unexpected default lifetime parameter",
-                                )
-                                .span_label(this.prev_token.span, "lifetime parameters cannot have default values")
-                                .emit();
-                           }
+                        if this.check_noexpect(&token::Eq) {
+                            if this.look_ahead(1, |t| t.is_lifetime()) {
+                                this.sess.emit_err(
```

Comment:
Sorry, I wasn't super clear when I gave my previous suggestion. We should still consume the default lifetime here:

```suggestion
                        if this.check_noexpect(&token::Eq) && this.look_ahead(1, |t| t.is_lifetime()) {
                                self.bump(2); // Consume `=` and `'lifetime`
                                this.sess.emit_err(
```

Reply:
Also, let's adjust the span here a bit:

Before bumping, let's get the span of the current token:

```
let lo = self.token.span;
self.bump(2);
let span = lo.to(self.prev_token.span);
```

Then use `span` in your error message. That should make the error message render like:

```
LL | pub struct DefaultLifetime<'a, 'b = 'static> {
   |                                   ^^^^^^^^^ lifetime parameters cannot have default values
```

I hope that makes sense.